### PR TITLE
EVG-19926: Upgrade go version to 1.20, and distros to latest.

### PR DESCRIPTION
**Pull request by johndaniels**

### CODE REVIEW

File: docs.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ logging, as well as systemd's journal service, and other logging
 systems, Grip provides a number of very powerful logging
 abstractions in one high-level package.
 
-Logging Instances
+# Logging Instances
```
Comment (Kimchelly) [inline docs.go:8]: I'm pretty sure godoc only recognizes plaintext and doesn't do HTML/Markdown formatting of any kind. Can we keep the bullet points?

**Review by Kimchelly — APPROVED** [review]
LGTM mod small thing.

File: docs.go (hunk 1 above)
Comment (johndaniels) [reply docs.go:8]: It recognizes a simple subset of markdown, ad as far as I know this doesn't remove any bullet points,

Comment (johndaniels) [reply docs.go:8]: gofmt insists on trying to turn these into headings and I didn't see a reason to fight it.

Comment (Kimchelly) [reply docs.go:8]: Interesting, alright then. Ignore the other comments about the same thing then.

File: evergreen.yaml [hunk 2]
```diff
@@ -131,47 +131,35 @@ tasks:
...
+      - ubuntu2204-small
     tasks: [ ".report" ]
 
   - name: ubuntu
-    display_name: Ubuntu 18.04
+    display_name: Ubuntu 22.04
     expansions:
-      GOROOT: /opt/golang/go1.16
+      GOROOT: /opt/golang/go1.20
+      RACE_DETECTOR: true
```
[the remark this answers is not in the document]
Comment (johndaniels) [reply evergreen.yaml:146]: I did it in all the ones I dropped arch linux from. It makes the tests slower but if we're going to gate on the race detector anyway in CI that seems like a non-concern.

**Pull request merged**